tests/apollo/util/bft_debug_tool.py

```diff
@@ -0,0 +1,150 @@
+                if child_proc.name() == 'skvbc_replica':
+                    ret = psutil.Process(child_proc.pid)
+                    self.external_tool_procs[started_replica_id] = started_proc_info   # external_tool_procs holds Heaptrack's process info
+                    break
+            assert started_replica_id in self.external_tool_procs
+            return ret
+        else:
+            return started_proc_info
+
+    def process_output(self, stopped_replica_id, stoppped_proc, testdir):
+        '''
+        Copy and/or extract and/or rename the debug tool logs into the correct output output folder
+        '''
+        output_log_format = "%s_skvbc_replica_pid_%d_repid_%d_#%d.%s" # tool name, process id, replica id, counter, postfix
+        if stopped_replica_id not in self.output_counters:
+            self.output_counters[stopped_replica_id] = 0    # May happen in reconfiguration
+        counter = self.output_counters[stopped_replica_id]
+        if self.name == "HEAPTRACK":
+            # Heaptrack output is a gz file. We would like to rename the gz file and extract it to a log file with heaptrack_print
+            assert (stopped_replica_id in self.external_tool_procs)
+            heaptrack_proc = self.external_tool_procs[stopped_replica_id]
+            heaptrack_proc.wait()
+            heaptrack_src_log_name = "%s%d%s" % ("heaptrack.skvbc_replica.", heaptrack_proc.pid, ".gz")
+            heaptrack_dst_base = output_log_format % (self.name.lower(), heaptrack_proc.pid, stopped_replica_id, counter, "%s")
+            heaptrack_dst_gz_path = os.path.join(self.output_log_path, heaptrack_dst_base % "gz")
+            heaptrack_dst_log_path = os.path.join(self.output_log_path, heaptrack_dst_base % "log")
+            shutil.copyfile(os.path.join(
+                testdir, heaptrack_src_log_name), os.path.join(self.output_log_path, heaptrack_dst_gz_path))
+            with open(heaptrack_dst_log_path, 'w') as logfile:
+                subprocess.call(["heaptrack_print","-f",heaptrack_dst_gz_path,"-t","1","-l","1"], stdout=logfile, shell=False)
+            del self.external_tool_procs[stopped_replica_id]
+        elif self.name in self.sanitizer_names:
+            # The default sanitizer output is <bft_network.output_log_prefix>.pid, e.g 'asan.log.170' where 170 is skvbc_replica pid
+            # We would like to rename the file, so that it contains the replica id as well, and a file counter per replica id.
+            pid = stoppped_proc.pid
+            sanitizer_src_file_path = os.path.join(self.output_log_path, "%s%s%d" %(self.output_log_prefix, ".", pid))
+            assert Path(sanitizer_src_file_path).is_file(), f"File {sanitizer_src_file_path} must exist!"
+            sanitizer_dst_file_path = os.path.join(self.output_log_path, \
+                output_log_format % (self.name.lower(), pid, stopped_replica_id, counter, "log"))
+            os.rename(sanitizer_src_file_path, sanitizer_dst_file_path)
```

Comment:
If you put Path(sanitizer_src_file_path) in a variable you can rename it using the Path object

Reply:
Thanks, I converted both source and dest paths to pathlib.Path and used rename .